src/Views/IframeView.php

```diff
@@ -1,88 +1,289 @@
+
+	/**
+	 * Form template.
+	 *
+	 * @var Template
+	 */
+	protected $template = null;
+
+	/**
+	 * Form id.
+	 *
+	 * @var int
+	 */
+	protected $formId = 0;
+
+	/**
+	 * IframeView Constructor
+	 *
+	 * @param Template $template
+	 */
+	public function __construct( $template = null ) {
+		$this->uniqueId    = uniqid( 'give-' );
+		$this->buttonTitle = __( 'Click to donate', 'give' );
+	}
+
+	/**
+	 * Set iframe URL.
+	 *
+	 * @param string $url
+	 *
+	 * @return $this
+	 */
+	public function setURL( $url = null ) {
+		$this->url = add_query_arg(
+			array_merge( [ 'giveDonationFormInIframe' => 1 ] ),
+			$url
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Set whether or not show modal.
+	 *
+	 * @param bool $set
+	 *
+	 * @return $this
 	 */
-	protected $bodyClasses = [ 'give-form-templates' ];
+	public function showInModal( $set ) {
+		$this->modal = $set;
+
+		return $this;
+	}
 
 	/**
-	 * Set document page title.
+	 * Button title.
 	 *
 	 * @param string $title
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setTitle( $title ) {
-		$this->title = $title;
+	public function setButtonTitle( $title ) {
+		$this->buttonTitle = $title;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Button color.
 	 *
-	 * @param string $body
+	 * @param string $color
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBody( $body ) {
-		$this->body = $body;
+	public function setButtonColor( $color ) {
+		$this->buttonColor = $color;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Form id.
 	 *
-	 * @param array $classes
+	 * @param int $id
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBodyClasses( $classes ) {
-		$this->bodyClasses = array_merge( $this->bodyClasses, (array) $classes );
+	public function setFormId( $id ) {
+		$this->formId = $id;
 
 		return $this;
 	}
 
+	/**
+	 * Get iframe HTML.
+	 *
+	 * @return string
+	 */
+	private function getIframeHTML() {
+		ob_start();
+		include $this->template->getLoadingView();
+
+		$loader = sprintf(
+			'<div class="iframe-loader">%1$s</div>',
+			ob_get_clean()
+		);
+
+		$iframe = sprintf(
+			'<iframe
+				name="give-embed-form"
+				%1$s
+				%4$s
+				data-autoScroll="%2$s"
+				onload="Give.initializeIframeResize(this)"
+				style="border: 0;visibility: hidden;%3$s"></iframe>%5$s',
+			$this->modal ? "data-src=\"{$this->url}\"" : "src=\"{$this->url}\"",
+			$this->autoScroll,
+			$this->minHeight ? "min-height: {$this->minHeight}px;" : '',
+			$this->modal ? 'class="in-modal"' : '',
+			$loader
+		);
+
+		if ( $this->modal ) {
+			$iframe = sprintf(
+				'<div class="modal-inner-wrap">
+					<div class="modal-content">
+		    			<a href="#" class="close-btn js-give-embed-form-modal-closer" aria-label="%3$s" data-form-id="%3$s" rel="nofollow">%2$s<span>&times;</span></a>
+						%1$s
+					</div>
+				</div>',
+				$iframe,
+				__( 'Close modal', 'give' ),
```

Comment:
Because translated strings are user-defined, the translated string should be escaped.